pyproject.toml

```
@@ -6,119 +6,190 @@ requires-python = ">=3.8"
-mantid = "*"  # Available in mantid/label/nightly channel
+snapred = ">=1.3.0"    # Available in neutrons channel
+mantid = "*"           # Available in mantid/label/nightly channel
 scikit-image = "*"
 python = ">=3.8"
 panel = ">=1.7.0"
 sqlalchemy = ">=2.0.0"
-versioningit = "*"
 numpy = "*"
 pandas = "*"
 matplotlib = "*"
 jupyterlab = "*"
 ipympl = "*"
+h5py = "*"
+
+# Environments and features
+
+[tool.pixi.environments]
+default = { features = [
+  "build",
+  "dev",
+  "docs",
+  "test",
+], solve-group = "default" }
+docs = { features = ["dev", "docs"], no-default-feature = true }
+build = { features = ["build"], no-default-feature = true }
+
+[tool.pixi.feature.build.dependencies]
+anaconda-client = ">=1.13.0"
+check-wheel-contents = ">=0.6.2"
+versioningit = ">=3.3.0"
+
+[tool.pixi.feature.build.pypi-dependencies]
+toml-cli = "*"
+
+[tool.pixi.feature.dev.dependencies]
+pip = "*"
+pip-audit = "*"
 pre-commit = "*"
-libmamba = "*"
-libarchive = "*"
-anaconda-client = "*"
-conda-build = "<4"
+
+[tool.pixi.feature.docs.dependencies]
 sphinx = "*"
 sphinx_rtd_theme = "*"
 myst-parser = "*"
+
+[tool.pixi.feature.test.dependencies]
 pytest = "*"
 pytest-cov = "*"
 pytest-xdist = "*"
-pip = "*"
-h5py = "*"
+
+[tool.pixi.tasks]
+# Documentation
+build-docs = { cmd = 'sphinx-build -b html docs docs/_build/html', description = "Build documentation" }
+test-docs = { cmd = "sphinx-build -M doctest docs docs/_build/html", description = "Test building the documentation" }
+# Testing
+test = { description = "Run the test suite", cmd = "pytest" }
+# Packaging
+conda-build-command = { cmd = "pixi build", description = "Wrapper for building the conda package - used by `conda-build`" }
+conda-build = { description = "Build the conda package", depends-on = [
+  "sync-version",
+  "conda-build-command",
+  "reset-version",
+] }
+conda-publish = { cmd = "anaconda upload *.conda", description = "Publish the .conda package to anaconda.org", depends-on = [
+  "conda-build",
+] }
+# Misc
+audit-deps = { cmd = "pip-audit --local -s osv", description = "Audit the package dependencies for vulnerabilities" }
+clean = { cmd = 'rm -rf .pytest_cache .ruff_cache **/*.egg-info **/dist **/__pycache__ **/_version.py', description = "Clean up various caches and build artifacts" }
+clean-conda = { cmd = "rm -f *.conda", description = "Clean the local .conda build artifacts" }
+clean-docs = { cmd = "rm -rf docs/_build", description = "Clean up documentation build artifacts" }
+clean-all = { description = "Clean all artifacts", depends-on = [
+  "clean",
+  "clean-conda",
+  "clean-docs",
+] }
+sync-version = { cmd = 'version=$(python -m versioningit); toml set tool.pixi.package.version "$version" --toml-path pyproject.toml', description = "Sync pyproject.toml version with Git version" }
+reset-version = { cmd = "toml set tool.pixi.package.version \"0.1.0\" --toml-path pyproject.toml", description = "Reset the package version to 0.1.0" }
+
```

I think all that's missing is a task to start workbench, since thats where users will typically be using snapwrap
```suggestion
workbench = { cmd = "python -m workbench", description = "Launch Workbench" }
```

And some updated docs.  @darshdinger could you add these?

---

does this need to be a task, or does it name sense to be "script" up top? (like `quicknxs-gui` for quicknxs)

---

I dont think so? Its not interacted with as a cli and there is nothing withing the package itself that explicitly launches a gui.  They use workbench, but need access to the one it is released with.

---

ohhhh, as in `mantidworkbench`?

---

Working on it now.